winpr/libwinpr/utils/image.c

```diff
@@ -125,66 +201,64 @@ int winpr_image_png_read_fp(wImage* image, FILE* fp)
 	}
 
 	lodepng_status = lodepng_decode32(&(image->data), &width, &height, data, size);
-
 	free(data);
 
 	if (lodepng_status)
 		return -1;
 
 	image->width = width;
 	image->height = height;
-
 	image->bitsPerPixel = 32;
 	image->bytesPerPixel = 4;
 	image->scanline = image->bytesPerPixel * image->width;
-
 	return 1;
 }
 
-int winpr_image_png_read_buffer(wImage* image, BYTE* buffer, int size)
+static int winpr_image_png_read_buffer(wImage* image, BYTE* buffer, size_t size)
 {
 	UINT32 width;
 	UINT32 height;
 	int lodepng_status;
-
 	lodepng_status = lodepng_decode32(&(image->data), &width, &height, buffer, size);
 
 	if (lodepng_status)
 		return -1;
 
 	image->width = width;
 	image->height = height;
-
 	image->bitsPerPixel = 32;
 	image->bytesPerPixel = 4;
 	image->scanline = image->bytesPerPixel * image->width;
-
 	return 1;
 }
 
-int winpr_image_bitmap_read_fp(wImage* image, FILE* fp)
+static int winpr_image_bitmap_read_fp(wImage* image, FILE* fp)
 {
+	int rc = -1;
 	int index;
 	BOOL vFlip;
 	BYTE* pDstData;
+	wStream* s;
 	WINPR_BITMAP_FILE_HEADER bf;
 	WINPR_BITMAP_INFO_HEADER bi;
+	s = Stream_New(NULL, sizeof(WINPR_BITMAP_FILE_HEADER) + sizeof(WINPR_BITMAP_INFO_HEADER));
 
-	if (fread((void*) &bf, sizeof(WINPR_BITMAP_FILE_HEADER), 1, fp) != 1)
+	if (!s)
 		return -1;
 
+	if (fread(Stream_Buffer(s), Stream_Capacity(s), 1, fp) != 1)
+		goto fail;
```

Comment:
This and the gotos below might cause a crash at ```free(image->data);``` because image->data is uninitialized here.